### DISCUSSION TITLE
optionalCatchAll errors when enabled

### DISCUSSION THREAD

**Discussion by Dannymx** [vercel/next.js#14808] [Help] [answered]
I wanted to try the `optionalCatchAll` for dynamic routes with my `index.js page`. Basically what I need is to access the following routes

```
http://localhost:3000/ -> no category is present, so this will show me all my items in the index page.
http://localhost:3000/category1 -> this will filter by items by category1
http://localhost:3000/category2 -> this will filter by items by category2
```
These three pages are pretty much the same, except they just filter down by category.

I enabled it like this in my `next.config.js`
```
const nextConfig = {
  experimental: {
    optionalCatchAll: true,
  },
};

module.exports = {
  nextConfig,
};
```
It works fine if I restart the server while my `index.js` is not renamed.

But as soon as I rename to `[[index]].js` and restart the server I get an error:
```
$ npm run dev

> frontend@0.1.0 dev C:\Users\Danny\Code\demo-strapi\frontend
> next dev

ready - started server on http://localhost:3000
info  - Loaded env from C:\Users\Danny\Code\demo-strapi\frontend\.env.local
> Using external babel configuration
> Location: "C:\Users\Danny\Code\demo-strapi\frontend\.babelrc"
event - compiled successfully
Error: Optional catch-all routes are currently experimental and cannot be used by default ("/[[...slug]]").
    at Watchpack.<anonymous> (C:\Users\Danny\Code\demo-strapi\frontend\node_modules\next\dist\server\next-dev-server.js:10:1513)
    at Watchpack.emit (events.js:310:20)
    at Watchpack._onTimeout (C:\Users\Danny\Code\demo-strapi\frontend\node_modules\watchpack\lib\watchpack.js:301:8)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
```

Is this normal? How this experimental feature should be enabled?

Here's my `.babelrc` just in case:
```
{
  "presets": ["next/babel"],
  "plugins": [
    [
      "import",
      {
        "libraryName": "antd",
        "libraryDirectory": "lib",
        "style": "index.css"
      }
    ]
  ]
}
```

**Comment by Timer** [comment, marked as answer]
Your `next.config.js` is invalid, try this:
```js
// next.config.js
const nextConfig = {
  experimental: {
    optionalCatchAll: true,
  },
};

module.exports = nextConfig;
```

**Reply by Dannymx** [reply]
Now I'm getting a 404 in `[[...index]].js`, when I try to access `http://localhost:3000/ale`

Below is my `getStaticPaths`, with the params commented as it used to work before.

```
export const getStaticPaths = async () => {
  return {
    // paths: [{ params: { index: "ale" } }, { params: { index: "lager" } }],
    paths: [
      {
        params: {
          index: ["ale", "lager"],
        },
      },
    ],
    fallback: false,
  };
};
```

**Reply by Timer** [reply]
Use the old commented out `paths` code. The new one you posted operates differently (expects `/ale/lager`).

**Reply by Dannymx** [reply]
But that's why I changed it to the other one, if I try the old one I get: `Error: A required parameter (index) was not provided as an array in getStaticPaths for /[[...index]]`, the new one got rid of the error but was giving me a 404.

**Reply by Timer** [reply]
Per the error message:
```js
paths: [{ params: { index: ["ale"] } }, { params: { index: ["lager"] } }],
```

Or easier:

```js
paths: ['/ale', '/lager'],
```

**Comment by tomgreeEn** [comment]
I'm getting this error message for Apollo when I enable in next.config.js:

Module not found: Can't resolve 'apollo/apolloClient.js' in '[path-to-project]/pages'

Using:
    "apollo-client": "^2.6.8",
    "next": "^9.3.5",

Here's my config file: 


`const withPlugins = require('next-compose-plugins');

const optimizedImages = require('next-optimized-images');

const path = require('path');

const nextConfig = {
  experimental: {
    optionalCatchAll: true
  }
};

module.exports = withPlugins(
  [
    [
      optimizedImages,
      {
        optimizeImagesInDev: true,
        handleImages: ['jpeg', 'png', 'svg', 'webp', 'gif', 'ico']
      }
    ]
  ],
  nextConfig,
  {
    webpack(config, options) {
      config.resolve.modules.push(path.resolve('./'));
      // config.plugins.push(new Dotenv({ silent: true }));
      return config;
    },
    env: {
      ALGOLIA_APP_ID: process.env.ALGOLIA_APP_ID,
      ALGOLIA_API_KEY: process.env.ALGOLIA_API_KEY,
      ROOT_URL: process.env.ROOT_URL
    }
    // poweredByHeader: false
  },
  {
    module: {
      loaders: [
        {
          test: /\.css$/,
          loaders: ['style?insertAt=top', 'css']
        }
      ]
    }
  }
);`
